kong/tools/batch_queue.lua

```diff
@@ -165,18 +166,19 @@ end
 -- @param self Queue
 -- @param batch: table with `entries` and `retries` counter
 -- @return nothing
-process = function(premature, self, batch)
+process = function(premature, self)
   if premature then
     return
   end
 
+  local batch = self.batch_queue[1]
```

Comment:
Removing the batch parameter and always using batch_queue[1] is certainly a good refactor, well done 👍